### DISCUSSION TITLE
Why do I get a 200 status response back when my route requires authentication?

### DISCUSSION THREAD

**Discussion by CallumVass** [pocketbase/pocketbase#1570] [Q&A] [answered]
Hi,

I have a route specified with the following ApiRule: @request.auth.id != ""

However, if I access that route without a token, I get a 200 back (albeit with no items returned which is good). But what I was expecting back was a 401 as I'm not authenticated?

In the logs I can see the auth is guest but I would still expect a 401 back if the route requires you to be authenticated?
![image](https://user-images.githubusercontent.com/2204920/211819725-76032f93-222e-47ea-9b0b-e6c52910eec2.png)

**Comment by ganigeorgiev** [comment, marked as answer]
This is because Collection API rules act also as filters and there is no easy and performant way to detect on which part of the rule the action will fail (for example you can also have a rule like `@request.auth.id != "" || status = 'public' ...`). The list/search action will return only the records that satisfies the rule. Similarly, you'll get 400 or 404 for the other actions in case the request client doesn't satisfy some of the other rules (view, create, update, delete).

You'll get 403 if the rule is marked as "Admin only" (aka. `null`).

**Reply by CallumVass** [reply]
Okay that makes sense, I guess there are no plans to implement a 401 response if the API rule is specified as `@request.auth.id != ""` and then the Auth is set to "guest"?

**Reply by ganigeorgiev** [reply]
No, sorry. As mentioned, API rules are also filters and while we can add an exception only for `@request.auth.id != ""` the rule can have other format or combination with other fields and the behavior will not be reliable or consistent.

**Reply by kisamoto** [reply]
@ganigeorgiev thank you for your explanation here. 

This tripped me up as well writing tests for my API rules as I was surprised that an unauthenticated user received a HTTP 200 which is why I came looking.

Could we update the docs to reflect this?

I'd be happy to have an attempt so if you create/have an issue for this please just tag me. 

Happy hacking
